cpp/tests/strings/contains_tests.cpp

```diff
@@ -460,6 +460,30 @@ TEST_F(StringsContainsTests, OverlappedClasses)
   }
 }
 
+TEST_F(StringsContainsTests, IncompleteClassesRange)
+{
+  auto input = cudf::test::strings_column_wrapper({"abc-def", "---", "", "ghijkl", "-wxyz-"});
+  auto sv    = cudf::strings_column_view(input);
+
+  {
+    auto results = cudf::strings::contains_re(sv, "[a-z]");
+    cudf::test::fixed_width_column_wrapper<bool> expected({1, 0, 0, 1, 1});
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
+  }
+  {
+    cudf::test::fixed_width_column_wrapper<bool> expected({1, 1, 0, 1, 1});
+    auto results = cudf::strings::contains_re(sv, "[g-]");
```

Comment:
I could imagine two other interesting test cases: hyphens being both a literal and a special character like `+--`, and multiple ranges `a-b-` or `-a-b`.

Also, it might make sense to test error cases like `a-b-c` explicitly? I guess we don't expect the second hyphen to be parsed as a literal in that case.

Reply:
Thanks Tobias. I added some more tests. 
Interestingly `[a-b-c]` is treated as `[a-c]` which I think makes sense and I like alot.
Looks like any multi-range [c<sub>1</sub>-c<sub>2</sub>-...-c<sub>n</sub>] is simply processed as [c<sub>1</sub>-c<sub>n</sub>] by the libcudf implementation.
However, this appears to be not consistent with other regex implementations.